### ISSUE TITLE
总的batchsize设置为4，训练120epoch结果与论文的结果差距较大

### ISSUE THREAD

**Issue by 15378695626** [fyw1999/MovingDroneCrowd#14]

**Comment by fyw1999** [comment]
> No description provided.

你跑的是SDNet在MovingDroneCrowd上的结果吗

**Comment by 15378695626** [comment]
是的呀



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2026年6月8日(周一) 晚上9:31
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [fyw1999/MovingDroneCrowd] 总的batchsize设置为4，训练120epoch结果与论文的结果差距较大 (Issue #14)


fyw1999 left a comment (fyw1999/MovingDroneCrowd#14)
  
No description provided.
  
你跑的是SDNet在MovingDroneCrowd上的结果吗

**Comment by fyw1999** [comment]
> 是的呀
> […](#)

划分文件设置的哪个，有使用预训练的密度图估计模型吗

**Comment by 15378695626** [comment]
划分文件设置的是：
TRAIN_LST = 'MDC_train.txt'
VAL_LST = 'MDC_val.txt'
TEST_LST = 'MDC_test.txt'
下面是配置文件：
import os
from easydict import EasyDict as edict
import time
# init
__C = edict()
cfg = __C


#------------------------------TRAIN------------------------
__C.SEED = 3035 &nbsp;# random seed, &nbsp;for reproduction
__C.MODEL = "SDNet" &nbsp;# model selection: GD3A, SDNet
__C.NAME = 'mdc_sdnet_conf_bs4_retry' &nbsp;# name of the experiment
__C.DATASET = 'MovingDroneCrowd' &nbsp; &nbsp; &nbsp; # dataset selection: &nbsp;HT21, SENSE
__C.encoder = "VGG16_FPN"
__C.RESUME = False # continue training
__C.RESUME_PATH = ''
__C.PRE_TRAINED_MODEL = ''
__C.GPU_ID = '4,5,6,7'
__C.USE_DEEPSPEED = False
__C.DEEPSPEED_CONFIG = ''
__C.USE_WANDB = True
__C.WANDB_PROJECT = 'MovingDroneCrowd-SDNet'
__C.WANDB_RUN_NAME = __C.NAME
__C.SEQ_SAVE_MAE_MAX = 50.0
__C.SEQ_SAVE_MSE_MAX = 80.0


# &nbsp; for GD3A
__C.global_counter = "STEERER"
__C.pre_trained_global_counter = ""
__C.sinkhorn_iterations = 100
__C.down_scale = 8
__C.ROI_RADIUS = 2.
__C.filter_thre = 0.005
__C.matched_thre = 0.1
__C.top_k = 9
__C.LR_Thre = 1e-4


# for SDNet
__C.PRE_TRAIN_COUNTER = '/home/xxx/thesis-code/vscode_projects/MovingDroneCrowd-main/quanzhong/SDNet_pre_trained_counter_MDC_VGG16_FPN_ep_200_downscale_16.pth'
__C.cross_attn_embed_dim = 256
__C.cross_attn_num_heads = 4
__C.mlp_ratio = 4
__C.cross_attn_depth = 2


# for both
__C.FEATURE_DIM = 256
__C.LR_Base = 1e-5 &nbsp;# learning rate
__C.WEIGHT_DECAY = 1e-6


__C.MAX_EPOCH = 120
__C.VAL_INTERVAL = 10
__C.START_VAL = 20
__C.PRINT_FREQ = 20
# print
now = time.strftime("%m-%d_%H-%M", time.localtime())


__C.EXP_NAME = now \
&nbsp; &nbsp; + '_' + __C.DATASET \
&nbsp; &nbsp; + '_' + str(__C.LR_Base) \
&nbsp; &nbsp; + '_' + __C.NAME


__C.EXP_PATH = os.path.join(
&nbsp; &nbsp; '/DATACENTER2/wenke.zhang/thesis-code/vscode_projects/MovingDroneCrowd-main/exp',
&nbsp; &nbsp; __C.DATASET
) &nbsp;# the path of logs, checkpoints, and current codes


os.makedirs(__C.EXP_PATH, exist_ok=True)

**Comment by fyw1999** [comment]
> 划分文件设置的是：
> TRAIN_LST = 'MDC_train.txt'
> VAL_LST = 'MDC_val.txt'
> TEST_LST = 'MDC_test.txt'
> 下面是配置文件：
> import os
> from easydict import EasyDict as edict
> import time
> # init
> __C = edict()
> cfg = __C
> 
> 
> #------------------------------TRAIN------------------------
> __C.SEED = 3035 &nbsp;# random seed, &nbsp;for reproduction
> __C.MODEL = "SDNet" &nbsp;# model selection: GD3A, SDNet
> __C.NAME = 'mdc_sdnet_conf_bs4_retry' &nbsp;# name of the experiment
> __C.DATASET = 'MovingDroneCrowd' &nbsp; &nbsp; &nbsp; # dataset selection: &nbsp;HT21, SENSE
> __C.encoder = "VGG16_FPN"
> __C.RESUME = False # continue training
> __C.RESUME_PATH = ''
> __C.PRE_TRAINED_MODEL = ''
> __C.GPU_ID = '4,5,6,7'
> __C.USE_DEEPSPEED = False
> __C.DEEPSPEED_CONFIG = ''
> __C.USE_WANDB = True
> __C.WANDB_PROJECT = 'MovingDroneCrowd-SDNet'
> __C.WANDB_RUN_NAME = __C.NAME
> __C.SEQ_SAVE_MAE_MAX = 50.0
> __C.SEQ_SAVE_MSE_MAX = 80.0
> 
> 
> # &nbsp; for GD3A
> __C.global_counter = "STEERER"
> __C.pre_trained_global_counter = ""
> __C.sinkhorn_iterations = 100
> __C.down_scale = 8
> __C.ROI_RADIUS = 2.
> __C.filter_thre = 0.005
> __C.matched_thre = 0.1
> __C.top_k = 9
> __C.LR_Thre = 1e-4
> 
> 
> # for SDNet
> __C.PRE_TRAIN_COUNTER = '/home/xxx/thesis-code/vscode_projects/MovingDroneCrowd-main/quanzhong/SDNet_pre_trained_counter_MDC_VGG16_FPN_ep_200_downscale_16.pth'
> __C.cross_attn_embed_dim = 256
> __C.cross_attn_num_heads = 4
> __C.mlp_ratio = 4
> __C.cross_attn_depth = 2
> 
> 
> # for both
> __C.FEATURE_DIM = 256
> __C.LR_Base = 1e-5 &nbsp;# learning rate
> __C.WEIGHT_DECAY = 1e-6
> […](#)
> 
> __C.MAX_EPOCH = 120
> __C.VAL_INTERVAL = 10
> __C.START_VAL = 20
> __C.PRINT_FREQ = 20
> # print
> now = time.strftime("%m-%d_%H-%M", time.localtime())
> 
> __C.EXP_NAME = now \
> &nbsp; &nbsp; + '_' + __C.DATASET \
> &nbsp; &nbsp; + '_' + str(__C.LR_Base) \
> &nbsp; &nbsp; + '_' + __C.NAME
> 
> __C.EXP_PATH = os.path.join(
> &nbsp; &nbsp; '/DATACENTER2/wenke.zhang/thesis-code/vscode_projects/MovingDroneCrowd-main/exp',
> &nbsp; &nbsp; __C.DATASET
> ) &nbsp;# the path of logs, checkpoints, and current codes
> 
> 
> os.makedirs(__C.EXP_PATH, exist_ok=True)

好的，我再重新训练一下试一下

**Comment by 15378695626** [comment]
感谢你重新验证实验，训练结束后烦请告知你的实验结果。
我严格依照论文与开源代码的配置复现，最终实验结果和原文存在偏差，对此我十分困惑。厘清这一问题，对保障该论文在ICCV上的可复现性也很有意义。



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2026年6月8日(周一) 晚上9:56
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [fyw1999/MovingDroneCrowd] 总的batchsize设置为4，训练120epoch结果与论文的结果差距较大 (Issue #14)


fyw1999 left a comment (fyw1999/MovingDroneCrowd#14)
  
划分文件设置的是：
 TRAIN_LST = 'MDC_train.txt'
 VAL_LST = 'MDC_val.txt'
 TEST_LST = 'MDC_test.txt'
 下面是配置文件：
 import os
 from easydict import EasyDict as edict
 import time
 
init
 
__C = edict()
 cfg = __C
 
#------------------------------TRAIN------------------------
 __C.SEED = 3035 &nbsp;# random seed, &nbsp;for reproduction
 __C.MODEL = "SDNet" &nbsp;# model selection: GD3A, SDNet
 __C.NAME = 'mdc_sdnet_conf_bs4_retry' &nbsp;# name of the experiment
 __C.DATASET = 'MovingDroneCrowd' &nbsp; &nbsp; &nbsp; # dataset selection: &nbsp;HT21, SENSE
 __C.encoder = "VGG16_FPN"
 __C.RESUME = False # continue training
 __C.RESUME_PATH = ''
 __C.PRE_TRAINED_MODEL = ''
 __C.GPU_ID = '4,5,6,7'
 __C.USE_DEEPSPEED = False
 __C.DEEPSPEED_CONFIG = ''
 __C.USE_WANDB = True
 __C.WANDB_PROJECT = 'MovingDroneCrowd-SDNet'
 __C.WANDB_RUN_NAME = __C.NAME
 __C.SEQ_SAVE_MAE_MAX = 50.0
 __C.SEQ_SAVE_MSE_MAX = 80.0
 
&nbsp; for GD3A
 
__C.global_counter = "STEERER"
 __C.pre_trained_global_counter = ""
 __C.sinkhorn_iterations = 100
 __C.down_scale = 8
 __C.ROI_RADIUS = 2.
 __C.filter_thre = 0.005
 __C.matched_thre = 0.1
 __C.top_k = 9
 __C.LR_Thre = 1e-4
 
for SDNet
 
__C.PRE_TRAIN_COUNTER = '/home/xxx/thesis-code/vscode_projects/MovingDroneCrowd-main/quanzhong/SDNet_pre_trained_counter_MDC_VGG16_FPN_ep_200_downscale_16.pth'
 __C.cross_attn_embed_dim = 256
 __C.cross_attn_num_heads = 4
 __C.mlp_ratio = 4
 __C.cross_attn_depth = 2
 
for both
 
__C.FEATURE_DIM = 256
 __C.LR_Base = 1e-5 &nbsp;# learning rate
 __C.WEIGHT_DECAY = 1e-6
 …
 
__C.MAX_EPOCH = 120
 __C.VAL_INTERVAL = 10
 __C.START_VAL = 20
 __C.PRINT_FREQ = 20
 
print
 
now = time.strftime("%m-%d_%H-%M", time.localtime())
 
__C.EXP_NAME = now 
 &nbsp; &nbsp; + '' + __C.DATASET 
 &nbsp; &nbsp; + '' + str(__C.LR_Base) 
 &nbsp; &nbsp; + '_' + __C.NAME
 
__C.EXP_PATH = os.path.join(
 &nbsp; &nbsp; '/DATACENTER2/wenke.zhang/thesis-code/vscode_projects/MovingDroneCrowd-main/exp',
 &nbsp; &nbsp; __C.DATASET
 ) &nbsp;# the path of logs, checkpoints, and current codes
 
os.makedirs(__C.EXP_PATH, exist_ok=True)
  
好的，我再重新训练一下试一下

**Comment by fyw1999** [comment]
> 感谢你重新验证实验，训练结束后烦请告知你的实验结果。
> 我严格依照论文与开源代码的配置复现，最终实验结果和原文存在偏差，对此我十分困惑。厘清这一问题，对保障该论文在ICCV上的可复现性也很有意义。
> […](#)

OK，得到结果我再告诉你，你也可以验证一下在MDC++上的效果，这些的结果的权重都有开源

**Comment by fyw1999** [comment]
The reproduction was conducted respectively under the global batch sizes of 4 and 8, and the results are as follows. Note that, to speed up the test, set `skip_flag` to `True`. As a result, the MIAE and MIOE results are inaccurate.
When global batch sizes is 8:

<img width="831" height="870" alt="Image" src="https://github.com/user-attachments/assets/43edb797-4335-4b5c-bbbb-919764daec0e" />

When global batch sizes is 4:

<img width="819" height="865" alt="Image" src="https://github.com/user-attachments/assets/175de994-6a00-4ef5-9838-212c30dfd0ec" />

When the global batch size is 8, some metrics (such as overall MAE) perform better than what was reported in the ICCV paper. The replicated results also align with the state-of-the-art performance of the competing algorithms at that time. SDNet is a weakly-supervised algorithm, which inherently has training instability and cannot guarantee that the results of each reproduction will be exactly the same. However, it can ensure that the SOTA performance of the comparison in the original paper. This instability is also one of the reasons for proposing the subsequent GD3A. If you want to reproduce the results of SDNet, it is best to ensure that the global batch size is 8 and use the pre-trained global counter.
The parameters for this reproduction are set as follows:
```python
import os
from easydict import EasyDict as edict
import time
# init
__C = edict()
cfg = __C

#------------------------------TRAIN------------------------
__C.SEED = 3035  # random seed,  for reproduction
__C.MODEL = "SDNet"  # model selection: GD3A, SDNet
__C.NAME = 'sdnet_test'  # name of the experiment
__C.DATASET = 'MovingDroneCrowd'       # dataset selection:  HT21, SENSE
__C.encoder = "VGG16_FPN"
__C.RESUME = False # continue training
__C.RESUME_PATH = ''
__C.PRE_TRAINED_MODEL = ''
__C.GPU_ID = '0,1,2,4'

#   for GD3A
__C.global_counter = "STEERER"
__C.pre_trained_global_counter = ""
__C.sinkhorn_iterations = 100
__C.down_scale = 8
__C.ROI_RADIUS = 2.
__C.filter_thre = 0.005
__C.matched_thre = 0.1
__C.top_k = 9
__C.LR_Thre = 1e-4

# for SDNet
__C.PRE_TRAIN_COUNTER = 'pre_train_model/SDNet_pre_trained_counter_MDC_VGG16_FPN_ep_200_downscale_16.pth'
__C.cross_attn_embed_dim = 256
__C.cross_attn_num_heads = 4
__C.mlp_ratio = 4
__C.cross_attn_depth = 2

# for both
__C.FEATURE_DIM = 256
__C.LR_Base = 1e-5  # learning rate
__C.WEIGHT_DECAY = 1e-6

__C.MAX_EPOCH = 120
__C.VAL_INTERVAL = 10
__C.START_VAL = 20
__C.PRINT_FREQ = 20
# print
now = time.strftime("%m-%d_%H-%M", time.localtime())

__C.EXP_NAME = now \
    + '_' + __C.DATASET \
    + '_' + str(__C.LR_Base) \
    + '_' + __C.NAME

__C.EXP_PATH = os.path.join('./exp', __C.DATASET)  # the path of logs, checkpoints, and current codes

os.makedirs(__C.EXP_PATH, exist_ok=True)
```
The settings for the relevant parameters of the dataset are as follows:
```python
from easydict import EasyDict as edict


# init
__C_MDC = edict()

cfg_data = __C_MDC

__C_MDC.TRAIN_SIZE = (768, 1024) # (848,1536) #
__C_MDC.TRAINING_MAX_LONG = 2560
__C_MDC.TRAINING_MAX_SHORT = 1440
__C_MDC.TEST_MAX_LONG = 1920
__C_MDC.TEST_MAX_SHORT = 1080
__C_MDC.DATA_PATH = '/data1/fyw/datasets/counting/MovingDroneCrowd++/'
__C_MDC.TRAIN_LST = 'MDC_train.txt'
__C_MDC.VAL_LST =  'MDC_val.txt'
__C_MDC.TEST_LST =  'MDC_test.txt'

__C_MDC.MEAN_STD = (
    [117/255., 110/255., 105/255.], [67.10/255., 65.45/255., 66.23/255.]
    # [0.4368163,  0.42558577, 0.41953506], [0.25696197, 0.25339108, 0.25411048]
)

__C_MDC.DEN_FACTOR = 200.

__C_MDC.TRAIN_BATCH_SIZE = 2 #  img pairs
__C_MDC.TRAIN_FRAME_INTERVALS=(3, 8)
__C_MDC.VAL_FRAME_INTERVALS = 4
__C_MDC.VAL_BATCH_SIZE = 1 # must be 1
```
Tips: Noted that during the reproduction process, both configurations achieved the best results on the validation set at the 120th epoch (best_model.pth). If you still cannot reproduce the results, it is recommended to further increase the number of training epochs (e.g., to 150) while ensuring that the global batch size is 8 and using the pre-trained global counter to further reduce the impact of instability training on the final results.

**Comment by fyw1999** [comment]
If you have any further questions, please feel free to contact me. Thank you for your attention. Additionally, I have uploaded the file [MDC_scene_labels.txt](https://huggingface.co/datasets/fyw1999/MovingDroneCrowd/resolve/main/MDC_scene_labels.txt) for reproducing the results of SDNet on MDC under different levels of difficulty. The current `scene_labels.txt` is used for MDC++, and MDC++ has made new partition for the levels of difficulty.

**Comment by 15378695626** [comment]
好的，辛苦了



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2026年6月10日(周三) 下午3:32
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [fyw1999/MovingDroneCrowd] 总的batchsize设置为4，训练120epoch结果与论文的结果差距较大 (Issue #14)


fyw1999 left a comment (fyw1999/MovingDroneCrowd#14)
 
Additionally, I have uploaded the file MDC_scene_labels.txt for reproducing the results of SDNet on MDC under different levels of difficulty. The current scene_labels.txt is used for MDC++, and MDC++ has made new partition for the levels of difficulty.
